cfep-XX.md

```diff
@@ -0,0 +1,61 @@
+Two cases are described depending on the status of the target name:
+
+- Unclaimed names: A package already present in conda-forge wants to use a new name currently unclaimed.
+- Names in use: A package, already present or about to be submitted, wishes to use a name already in use.
+
+## Specification
+
+This CFEP follows [RFC 2119](https://datatracker.ietf.org/doc/html/rfc2119) in its use of certain keywords (e.g., MUST).
+
+### Feedstock Epochs
+
+In order to track how feedstocks are reused, we define the concept of the `feedstock epoch` and a `feedstock epoch label`.
+
+- When a feedstock is first created, its epoch is 0.
+- If a feedstock is reused for a new project (see below), the `feedstock epoch` MUST be incremented by 1.
+- A `feedstock epoch label` is a label of the form `{feedstock name without '-feedstock'}_feedstock_epoch{N}`. When the label is applied to an artifact, the label MUST reflect epoch of the feedstock at the time it produced the given artifact.
+- All artifacts uploaded from a feedstock which do not otherwise have a `feedstock epoch label` are assumed to be from the current `feedstock epoch`.
+- A `feedstock epoch label` MUST be applied to any artifact from the previous epoch of the feedstock.
+- The current `feedstock epoch` for a given feedstock is determined as follows. The set of all labels for all artifacts produced by the feedstock is searched for any label matching the form of a `feedstock epoch label` for that feedstock. If no label matches, then the current `feedstock epoch` is 0. If labels do match, then the current `feedstock epoch` is one plus the maximum `feedstock epoch` over all matching labels.
+
+### Unclaimed names
+
+When an existing package wants to be renamed to a currently unclaimed name, the process is uncontroversial and straight-forward. Two approaches are possible:
+
+- The new name needs to be registered to the originating feedstock in `feedstock-outputs` (or equivalent). Once approved, a PR in the feedstock adding the new output may proceed normally. The old name MUST be kept as an alias output for backwards compatibility. In this case, the feedstock retains the name of the original name.
+- The original feedstock is archived and the package is re-submitted to `staged-recipes` (or equivalent) under the new name. The old package name MUST be added as an alias output for backwards compatibility, and MUST be registered as a valid output of the new feedstock in `feedstock-outputs`.
+
+### Names in use
+
+The blanket policy in this case is that an existing package name may not be used by a different project. In exceptional cases, the old name may be reallocated provided that:
+
+- The old project is abandoned. Factors that can help to indicate if a project is abandoned include, but are not limited to, the project's git repo being archived, the project's distributions no longer being available (e.g., yanked artifacts on PyPI), etc.
```

Comment:
I think we can and probably should explicitly gate this on whether the name has been "inherited" on PyPI (where applicable). We gain nothing IMO from creating gratuitous divergence by handling the situation differently than PyPI (in terms of different behaviours between `pip install foo` and `conda install foo`, and if we did, the name mapping between PyPI and conda would then also take on a temporal component)